lib/Parse/ParseType.cpp

```diff
@@ -41,23 +41,12 @@ TypeRepr *Parser::applyAttributeToType(TypeRepr *ty,
     ty = new (Context) AttributedTypeRepr(attrs, ty);
   }
 
-  // Apply 'inout' or '__shared' or '__owned'
-  if (specifierLoc.isValid()) {
-    switch (specifier) {
-    case ParamDecl::Specifier::Owned:
-      ty = new (Context) OwnedTypeRepr(ty, specifierLoc);
-      break;
-    case ParamDecl::Specifier::InOut:
-      ty = new (Context) InOutTypeRepr(ty, specifierLoc);
-      break;
-    case ParamDecl::Specifier::Shared:
-      ty = new (Context) SharedTypeRepr(ty, specifierLoc);
-      break;
-    case ParamDecl::Specifier::Default:
-      break;
-    }
+  // Apply 'inout', 'consuming', or 'borrowing' modifiers.
+  if (specifierLoc.isValid() &&
+      specifier != ParamDecl::Specifier::Default) {
+    ty = new (Context) OwnershipTypeRepr(ty, specifier, specifierLoc);
```

Comment:
would it be valid to add an `assert(specifier != 0)` in the constructor of `OwnershipTypeRepr`? I guess maybe you're just trying to avoid wrapping the type needlessly.